### PR TITLE
Set pin.valist.io as default backup pinning provider

### PR DESCRIPTION
**Pull request by jiyuu-jin**

### CODE REVIEW

File: storage/pinning/pinning.go [hunk 1]
```diff
@@ -60,12 +60,36 @@ func (prov *Provider) Write(ctx context.Context, data []byte) (string, error) {
...
+		if err != nil {
+			pw.CloseWithError(err) //nolint:errcheck
+			return
+		}
+	}()
+
+	defer pr.Close()
+
+	req, err := http.NewRequest(http.MethodPost, prov.host+"/api/v0/dag/import", pr)
+	req.Header.Set("Content-Type", mw.FormDataContentType())
```
Comment (nasdf) [inline storage/pinning/pinning.go:88]: Request could be nil here, so you should do the error check first then set the header.

File: storage/pinning/pinning.go [hunk 2]
```diff
@@ -60,12 +60,36 @@ func (prov *Provider) Write(ctx context.Context, data []byte) (string, error) {
...
 
-	req, err := http.NewRequest(http.MethodPost, prov.host+"/content/add-car", car)
+	// use a pipe to stream request data
+	pr, pw := io.Pipe()
+	// write form data to the pipe writer
+	mw := multipart.NewWriter(pw)
+	// pipe reader will block until content is written to pipe writer
+	// this is how the data is able to be streamed in a routine
+	go func() {
+		defer mw.Close()
```
Comment (nasdf) [inline storage/pinning/pinning.go:70]: The writer is never properly closed. This should fix it.

```suggestion
                 // create a form file for the multipart data
		ff, err := mw.CreateFormFile("path", strings.TrimPrefix(fpath, "/ipfs/"))
		if err != nil {
			pw.CloseWithError(err) //nolint:errcheck
			return
		}
		// copy the input data to the form file
		_, err = io.Copy(ff, car)
		if err != nil {
			pw.CloseWithError(err) //nolint:errcheck
			return
		}
		
		err = mw.Close()
		pw.CloseWithError(err) //nolint:errcheck
```

File: storage/pinning/pinning.go [hunk 3]
```diff
@@ -60,12 +60,36 @@ func (prov *Provider) Write(ctx context.Context, data []byte) (string, error) {
...
+		}
+		// copy the input data to the form file
+		_, err = io.Copy(ff, car)
+		if err != nil {
+			pw.CloseWithError(err) //nolint:errcheck
+			return
+		}
+	}()
+
+	defer pr.Close()
```
Comment (nasdf) [inline storage/pinning/pinning.go:85]: Shouldn't need to close the read side since the write side is closed in the go routine.

Comment (jiyuu-jin) [reply storage/pinning/pinning.go:85]: Awesome, removed line 85.

File: storage/pinning/pinning.go (hunk 1 above)
Comment (jiyuu-jin) [reply storage/pinning/pinning.go:102]: Great catch, I moved the error check to be first.

**Review by awantoch — APPROVED** [review]

**Pull request merged**